plugins/com.google.cloud.tools.eclipse.appengine.facets/plugin.xml

```diff
@@ -113,15 +113,33 @@
   <!-- begin App Engine flex facet -->
   <extension point="org.eclipse.wst.common.project.facet.core.facets">
     <project-facet id="com.google.cloud.tools.eclipse.appengine.facets.flex">
-       <label>%flexFacetName</label>
-       <description>%flexFacetDescription</description>
+      <label>%flexFacetName</label>
+      <description>%flexFacetDescription</description>
     </project-facet>
     <project-facet-version facet="com.google.cloud.tools.eclipse.appengine.facets.flex" version="1">
-       <constraint>
+      <constraint>
+        <and>
           <conflicts facet="com.google.cloud.tools.eclipse.appengine.facets.standard"/>
-        </constraint>
+          <requires group="com.google.cloud.tools.eclipse.appengine.facets.flex.constraints.servlet"/>
```

Comment:
I'm regretting this approach for the Java 8 support.  Although groups are extensible, they don't allow specifying additional constraints: it's just "at least one from this group".